lib/account/pages/write_review_page.dart

```diff
@@ -148,32 +149,49 @@ class _WriteReviewPageState extends State<WriteReviewPage> {
   }
 
   void onRatingUpdate(double rating) {
+    reviewChanged = true;
     setState(() {
       _review!.rating = rating.toInt();
     });
   }
 
+  void _updateOverallReview() {
+    if (!reviewChanged) {
+      _review!.rating = ((_review!.comfortRating ?? 0) +
```

Comment:
I think this code could be improved in readability.
Something like
```
_review!.rating = <int?>[_review!.comfortRating, _review!.safetyRating, _review!.hospitalityRating,
_review!.reliabilityRating].where((int? element) => element != null).average;
```

Reply:
Btw I wasn't kidding with average (part of the collection package). Here is working code:

```
      
final List<int> givenCategoryRatings = <int?>[
        _review!.comfortRating,
        _review!.safetyRating,
        _review!.hospitalityRating,
        _review!.reliabilityRating
      ].whereType<int>().toList();
_review!.rating = givenCategoryRatings.average.round();
```